compiler/desugared/from_surface.ml

```diff
@@ -366,7 +375,7 @@ let rec translate_expr
           (try
              Runtime.date_of_numbers date.literal_date_year
                date.literal_date_month date.literal_date_day
-           with Runtime.ImpossibleDate ->
+           with Dates_calc.Dates.InvalidDate ->
```

Comment:
In my mind the Catala compiler would never directly depend on `Dates_calc`, to ensure some sort of abstraction of how things like dates are actually implemented. Which is why the runtime re-exported this error and why we used it here. Do you have a good rationale for tweaking this?

Reply:
Ok, no indeed you have a good point, I'll ensure it's properly shadowed